index.html

```diff
@@ -25,43 +25,61 @@ <h1>Hello Plugins!</h1>
     </details>
     <br/>
 
+    <div class="pk-container" style="margin-bottom: 10px">
+      <label class="">Public key:</label>
+      <div class="pk"></div>
+    </div>
     <button class="connect">Connect</button>
-    <button class="generateKeys">Generate keys</button>
+
   </body>
 
   <script>
 
-    const origin = new URL('package.json', window.location.href).toString()
-    const snapId = `wallet_plugin_${origin}`
+    const origin = new URL('package.json', window.location.href).toString();
+    const snapId = `wallet_plugin_${origin}`;
 
-    const connectButton = document.querySelector('button.connect')
-    const generateKeysButton = document.querySelector('button.generateKeys')
+    const connectButton = document.querySelector('button.connect');
+    const pkLabel = document.querySelector('div.pk');
 
-    connectButton.addEventListener('click', connect)
-    generateKeysButton.addEventListener('click', send)
+    connectButton.addEventListener('click', connect);
+    connectButton.disabled = true; // until keypair is fetched
 
-    async function connect () {
-      await ethereum.send({
-        method: 'wallet_requestPermissions',
-        params: [{
-          [snapId]: {}
-        }]
-      })
-    }
+    fetchPublicKey().then((fetched) => {
```

Comment:
Why do you need this? You are calling this after connect

Reply:
In case you already connected snap and have saved public key in a state.